src/clangd-context.ts

```diff
@@ -122,6 +122,20 @@ export class ClangdContext implements vscode.Disposable {
             // notice until the behavior was in several releases, so we need
             // to override it on the client.
             item.commitCharacters = [];
+            // VSCode won't automatically trigger signature help when entering
+            // a placeholder, e.g. if the completion inserted brackets and
+            // placed the cursor inside them.
+            // https://github.com/microsoft/vscode/issues/164310
+            // They say a plugin should trigger this, but LSP has no mechanism.
+            // https://github.com/microsoft/language-server-protocol/issues/274
+            // (This workaround is incomplete, and only helps the first param).
+            if (item.insertText instanceof vscode.SnippetString &&
```

Comment:
Is it intentional that the workaround is not reached if the user has set `"clangd.serverCompletionRanking": false` (which causes us to early-return near the top of this function)?

Reply:
(The likely answer here is that there's no reason the workaround couldn't also apply to the `"clangd.serverCompletionRanking": false` case, we'd just have to reorganize the code a bit to handle that case as well. We can do that in a follow-up patch if anyone asks for it.)